### PR TITLE
Deploying Cloud Services (extended support)

### PR DESCRIPTION
**Pull request by gauravchandwani1**
# PR Checklist

Check these items before submitting a PR... 

[Contribution Guide](https://github.com/Azure/azure-quickstart-templates/blob/master/1-CONTRIBUTION-GUIDE/README.md)

[Best Practice Guide](https://github.com/Azure/azure-quickstart-templates/blob/master/1-CONTRIBUTION-GUIDE/best-practices.md)

- [x] - Please check this box once you've submitted the PR if you've read through the Contribution Guide and best practices checklist.

## Changelog

*
*
*

### CODE REVIEW

**Comment by mimckitt** [comment]
Following @mimckitt

**Comment by gauravchandwani1** [comment]
Create cloud service (extended support

**Comment by gauravchandwani1** [comment]
Closed by mistake.

File: 101-cses-multi-role-rdp/azuredeploy.json [hunk 1]
```diff
@@ -0,0 +1,180 @@
+{
+  "$schema": "https://schema.management.azure.com/schemas/2015-01-01/deploymentTemplate.json#",
+  "contentVersion": "1.0.0.0",
+  "parameters": {
+    "cloudServiceName": {
+      "type": "string",
```
Comment (bmoore-msft) [inline 101-cses-multi-role-rdp/azuredeploy.json:6]: add a defaultValue using  uniqueString() - also mention in the description that this must be globally unique

File: 101-cses-multi-role-rdp/azuredeploy.json [hunk 2]
```diff
@@ -0,0 +1,180 @@
...
+      "metadata": {
+        "description": "Name of the cloud service (extended support)"
+      }
+    },
+    "location": {
```
Comment (bmoore-msft) [inline 101-cses-multi-role-rdp/azuredeploy.json:11]: Add defaultValue - see: https://github.com/Azure/azure-quickstart-templates/blob/master/1-CONTRIBUTION-GUIDE/best-practices.md#parameters

File: 101-cses-multi-role-rdp/azuredeploy.json [hunk 3]
```diff
@@ -0,0 +1,180 @@
...
+      "type": "string",
+      "metadata": {
+        "description": "Location of the cloud service (extended support)"
+      }
+    },
+    "deploymentLabel": {
```
Comment (bmoore-msft) [inline 101-cses-multi-role-rdp/azuredeploy.json:17]: add defaultValue

File: 101-cses-multi-role-rdp/azuredeploy.json [hunk 4]
```diff
@@ -0,0 +1,180 @@
...
+      "type": "string",
+      "metadata": {
+        "description": "Label of the deployment"
+      }
+    },
+    "packageSasUri": {
```
Comment (bmoore-msft) [inline 101-cses-multi-role-rdp/azuredeploy.json:23]: See [BP guide](https://github.com/Azure/azure-quickstart-templates/blob/master/1-CONTRIBUTION-GUIDE/best-practices.md#deployment-artifacts-nested-templates-scripts) for how to handle artifacts

File: 101-cses-multi-role-rdp/azuredeploy.json [hunk 5]
```diff
@@ -0,0 +1,180 @@
...
+      }
+    },
+    "configurationSasUri": {
+      "type": "securestring",
+      "metadata": {
+        "description": "SAS Uri of the service configuration (.cscfg)"
+      }
+    },
+    "roles": {
+      "type": "array",
```
Comment (bmoore-msft) [inline 101-cses-multi-role-rdp/azuredeploy.json:36]: add a defaultValue

File: 101-cses-multi-role-rdp/azuredeploy.json [hunk 6]
```diff
@@ -0,0 +1,180 @@
...
+    "upgradeMode": {
+      "type": "string",
+      "defaultValue": "Auto",
+      "metadata": {
+        "UpgradeMode": "UpgradeMode of the CloudService"
+      }
+    }
+  },
+  "variables": {
+    "cloudServiceName": "[parameters('cloudServiceName')]",
```
Comment (bmoore-msft) [inline 101-cses-multi-role-rdp/azuredeploy.json:76]: remove this and just use the parameter directly

File: 101-cses-multi-role-rdp/azuredeploy.json [hunk 7]
```diff
@@ -0,0 +1,180 @@
...
+    "subscriptionID": "[subscription().subscriptionId]",
+    "dnsName": "[variables('cloudServiceName')]",
+    "lbName": "[concat(variables('cloudServiceName'), 'LB')]",
+    "lbFEName": "[concat(variables('cloudServiceName'), 'LBFE')]",
+    "resourcePrefix": "[concat('/subscriptions/', variables('subscriptionID'), '/resourceGroups/', resourceGroup().name, '/providers/')]"
```
Comment (bmoore-msft) [inline 101-cses-multi-role-rdp/azuredeploy.json:81]: remove this, you will need to use the resourceId function for resourceIds

File: 101-cses-multi-role-rdp/azuredeploy.json [hunk 8]
```diff
@@ -0,0 +1,180 @@
...
+      "apiVersion": "2020-10-01-preview",
+      "type": "Microsoft.Compute/cloudServices",
+      "name": "[variables('cloudServiceName')]",
+      "location": "[parameters('location')]",
+      "tags": {
+        "DeploymentLabel": "[parameters('deploymentLabel')]",
+        "DeployFromVisualStudio": "true"
+      },
+      "dependsOn": [
+        "[concat('Microsoft.Network/virtualNetworks/', parameters('vnetName'))]",
```
Comment (bmoore-msft) [inline 101-cses-multi-role-rdp/azuredeploy.json:132]: remove concat() and use resourceId

File: 101-cses-multi-role-rdp/azuredeploy.json [hunk 9]
```diff
@@ -0,0 +1,180 @@
...
+        "packageUrl": "[parameters('packageSasUri')]",
+        "configurationUrl": "[parameters('configurationSasUri')]",
+        "upgradeMode": "[parameters('upgradeMode')]",
+        "roleProfile": {
+          "roles": "[parameters('roles')]"
+        },
+        "networkProfile": {
+          "loadBalancerConfigurations": [
+            {
+              "id": "[concat(variables('resourcePrefix'), 'Microsoft.Network/loadBalancers/', variables('lbName'))]",
```
Comment (bmoore-msft) [inline 101-cses-multi-role-rdp/azuredeploy.json:145]: remove concat() and use resourceId()

File: 101-cses-multi-role-rdp/azuredeploy.json [hunk 10]
```diff
@@ -0,0 +1,180 @@
...
+              "name": "[variables('lbName')]",
+              "properties": {
+                "frontendIPConfigurations": [
+                  {
+                    "name": "[variables('lbFEName')]",
+                    "properties": {
+                      "publicIPAddress": {
+                        "id": "[concat(variables('resourcePrefix'), 'Microsoft.Network/publicIPAddresses/', parameters('publicIPName'))]"
```
Comment (bmoore-msft) [inline 101-cses-multi-role-rdp/azuredeploy.json:153]: remove concat() and use resourceId()

File: 101-cses-multi-role-rdp/azuredeploy.parameters.json [hunk 11]
```diff
@@ -0,0 +1,45 @@
+{
+  "$schema": "https://schema.management.azure.com/schemas/2015-01-01/deploymentParameters.json#",
+  "contentVersion": "1.0.0.0",
+  "parameters": {
+    "cloudServiceName": {
```
Comment (bmoore-msft) [inline 101-cses-multi-role-rdp/azuredeploy.parameters.json:5]: remove the parameters that have defaultValues

File: 101-cses-multi-role-rdp/azuredeploy.parameters.json [hunk 12]
```diff
@@ -0,0 +1,45 @@
...
+        {
+          "name": "WorkerRole1",
+          "sku": {
+            "name": "Standard_D1_v2",
+            "capacity": "1"
+          }
+        }
+      ]
+    },
+    "rdpPublicConfig": {
```
Comment (bmoore-msft) [inline 101-cses-multi-role-rdp/azuredeploy.parameters.json:38]: for the config (and private config) - move the XML to a variable in the file and prompt for username and password as the param values.  Can also remove the \r\n

Asking the user to supply XML within JSON will be prone to failure.

File: 101-cses-multi-role-rdp/metadata.json [hunk 13]
```diff
@@ -0,0 +1,12 @@
+{
+  "$schema": "https://aka.ms/azure-quickstart-templates-metadata-schema#",
+  "type": "QuickStart",
+  "itemDisplayName": "Blank Template",
+  "description": "Cloud Services (extended support) multi-role deployment with RDP",
+  "summary": "This template allows deploying a Cloud Services (extended support) with mulitple roles. It also creates virtual network, public ip address.",
```
Comment (bmoore-msft) [inline 101-cses-multi-role-rdp/metadata.json:6]: This template deploys an Azure Cloud Service (extended support) with multiple roles. It also creates a virtual network and public ip address.

File: 101-cses-multi-role-rdp/readme.md [hunk 14]
```diff
@@ -0,0 +1,17 @@
+# Cloud Services (extended support) template
```
Comment (bmoore-msft) [inline 101-cses-multi-role-rdp/readme.md:1]: change the casing on the filename to: README.md

Also see the pipeline results for the proper links

File: 101-cses-multi-role-rdp/readme.md [hunk 15]
```diff
@@ -0,0 +1,17 @@
...
+
+[![Deploy To Azure](https://raw.githubusercontent.com/Azure/azure-quickstart-templates/master/1-CONTRIBUTION-GUIDE/images/deploytoazure.svg?sanitize=true)](https://portal.azure.com/#create/Microsoft.Template/uri/https%3A%2F%2Fraw.githubusercontent.com%2FAzure%2Fazure-quickstart-templates%2Fmaster%2F100-blank-template%2Fazuredeploy.json)
+[![Deploy To Azure US Gov](https://raw.githubusercontent.com/Azure/azure-quickstart-templates/master/1-CONTRIBUTION-GUIDE/images/deploytoazuregov.svg?sanitize=true)](https://portal.azure.us/#create/Microsoft.Template/uri/https%3A%2F%2Fraw.githubusercontent.com%2FAzure%2Fazure-quickstart-templates%2Fmaster%2F100-blank-template%2Fazuredeploy.json)
+[![Visualize](https://raw.githubusercontent.com/Azure/azure-quickstart-templates/master/1-CONTRIBUTION-GUIDE/images/visualizebutton.svg?sanitize=true)](http://armviz.io/#/?load=https%3A%2F%2Fraw.githubusercontent.com%2FAzure%2Fazure-quickstart-templates%2Fmaster%2F100-blank-template%2Fazuredeploy.json)
+
+This template allows to create cloud service (extended support) and also creates associated resources (virtual network, public IP address).
```
Comment (bmoore-msft) [inline 101-cses-multi-role-rdp/readme.md:13]: This template deploys an Azure Cloud Service (extended support) with multiple roles. It also creates a virtual network and public ip address.

**Review by bmoore-msft — CHANGES_REQUESTED** [review]

**Comment by bmoore-msft** [comment]
Closing abandoned request - please reopen if you can address the feedback.

**Comment by gauravchandwani1** [comment]
Surbhi - Please look into this.

**Comment by gachandw** [comment]
#sign-off

**Pull request closed without merging**